City-Bureau/city-scrapers-indianapolis
# 🏗️ Build spider: Indianapolis City-County Council

## What's this PR do?
Adds a spider to scrape meetings from the website of the Indianapolis City-County Council. The new spider is called "ind_city_county".

## Why are we doing this?
Requested by our partners in Indianapolis.

## Steps to manually test

1. Ensure the project is installed:
```
pipenv sync --dev
```
2. Activate the virtual env and enter the pipenv shell:
```
pipenv shell
```
3. Run the spider:
```
scrapy crawl ind_city_county -O test_output.csv
```
4. Monitor the output and ensure no errors are raised.

5. Inspect `test_output.csv` to ensure the data looks valid. Target website is [here](https://calendar.indy.gov/). Change the "view type" to list (the leftmost option) for a more helpful view of the meetings.

## Are there any smells or added technical debt to note? 
- I note this in the docstring of the spider's `parse` method but I'll also flag here: this agency includes many events that are not meetings in its calendar. I opted to exclude any meetings that can't be neatly categorized as a board, committee, commission or council meeting based on its title. Most of these non-meeting events are community events like charitable giving days or anniversary days. The risk of this filtering is that we accidentally ignore meetings that might not include board/committee/commission/council keywords in their titles but I think it's likely worth it to avoid the noise. As of today, without this filtering, the spider would return 125 events. With the parsing, the returned events are cut down to 53 – meaning more than 50% of the events would be irrelevant.

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1206555349401751
